src/shogun/preprocessor/RandomFourierGaussPreproc.h

```diff
@@ -91,147 +61,68 @@ class RandomFourierGaussPreproc: public RandomMixin<DensePreprocessor<float64_t>
+	 * @return feature space dimension
 	 */
-	int32_t get_dim_input_space() const;
-
-	/** a getter
-	 * @return the set value of protected member dim_feature_space
-	 */
-	int32_t get_dim_feature_space() const;
+	int32_t get_dim_feature_space() const noexcept;
 
 	/** inherited from base class
 	 * does nothing
 	 */
 	void cleanup();
 
-	/// return the name of the preprocessor
 	virtual const char* get_name() const { return "RandomFourierGaussPreproc"; }
 
-	/// return a type of preprocessor
 	virtual EPreprocessorType get_type() const { return P_RANDOMFOURIERGAUSS; }
 
 protected:
 	/** default processing routine, inherited from base class
 	 * @param matrix the features matrix to be processed
-	 * @return the processed feature matrix from the DenseFeatures<float64_t>
-	 * class in case (2) (see description above) this routine requires only
-	 * steps 2a) and 2b), the rest is determined automatically
+	 * @return the processed feature matrix of class DenseFeatures<float64_t>
 	 */
 	virtual SGMatrix<float64_t> apply_to_matrix(SGMatrix<float64_t> matrix);
 
-	/**
-	 * helper for copy constructor and assignment operator=
-	 */
-	void copy(const RandomFourierGaussPreproc & feats); // helper for two constructors
-
-
-	/** dimension of input features
-	 * width of gaussian kernel in the form of exp(-x^2 / (2.0 kernelwidth^2) ) NOTE the 2.0 and the power ^2 !
-	 */
-	float64_t kernelwidth;
-
-	/** dimension of input features
-	 * width of gaussian kernel in the form of exp(-x^2 / (2.0 kernelwidth^2) ) NOTE the 2.0 and the power ^2 !
-	 */
-	float64_t cur_kernelwidth;
-
-	/** desired dimension of input features as set by void set_dim_input_space(const int32_t dim)
-	 *
+	/** Samples from the fourier transform of the kernel and stores into
+	 * the internal members, inherited from base class
+	 * @param dim_input_space input space dimension
 	 */
-	int32_t dim_input_space;
+	virtual void sample_spectral_density(int32_t dim_input_space);
 
-	/** actual dimension of input features as set by bool init_randomcoefficients() or void set_randomcoefficients
+	/** Helper method which generates random coefficients and stores in the
+	 *  internal members. This function is called by fit() which does the error
+	 *  checking before passing here, inherited from base class
 	 *
+	 * @param dim_input_space input space dimension
 	 */
-	int32_t cur_dim_input_space;
+	virtual void init_basis(int32_t dim_input_space);
 
+	/** width of gaussian kernel in the form exp(-x^2 / (2.0* kernelwidth^2) )*/
+	float64_t m_kernelwidth;
 
-	/** desired dimension of output features  as set by void set_dim_feature_space(const int32_t dim)
-	 *
-	 */
-	int32_t dim_feature_space;
-
-	/** actual dimension of output features as set by bool init_randomcoefficients() or void set_randomcoefficients
-	 *
-	 */
-	int32_t cur_dim_feature_space;
-
-	/**
-	 * tests whether rf features have already been initialized
-	 */
-	bool test_rfinited() const;
+	/** desired dimension of output features */
```

Comment:
as mentioned before, a better var name would help and then the docstring could be removed `m_dim_output` or `m_num_rff`